### PR TITLE
[Android] Default to auto width for non valid width values

### LINKED ISSUE #3864: [Android][Inconsistency with web and iOS] [Invalid value for column width not defaulting to auto in Android]
# Platform

What platform is your issue or question related to? (Delete other platforms).
* Android
# Author or host
HOST
# Version of SDK
1.2.6
# Details
Consider the following json
`{
               "type":"AdaptiveCard",
               "version":"1.2",
               "$schema":"http://adaptivecards.io/schemas/adaptive-card.json",
               "body":[
                  {
                     "type":"ColumnSet",
                     "spacing":"none",
                     "columns":[
                        {
                           "type":"Column",
                           "items":[
                              {
                                 "type":"Image",
                                 "url":"https://www.sewausa.org/resources/Sewa%20USA%20Images/DIY%20Campaigns/profile-clipart-generic-user-12.png",
                                 "horizontalAlignment":"left",
                                 "size":"small",
                                 "style":"person"
                              }
                           ],
                           "horizontalAlignment":"left",
                           "width":"automatic",
                           "verticalContentAlignment":"center",
                           "separator":false
                        },
                        {
                           "type":"Column",
                           "verticalContentAlignment":"center",
                           "items":[
                              {
                                 "type":"TextBlock",
                                 "text":"Some text here",
                                 "weight":"Bolder"
                              },
                              {
                                 "type":"TextBlock",
                                 "text":" "
                              }
                           ]
                        },
                        {
                           "type":"Column",
                           "verticalContentAlignment":"center",
                           "items":[
                              {
                                 "type":"ActionSet",
                                 "actions":[
                                    {
                                       "type":"Action.Submit",
                                       "title":"Button description",
                                       "data":{
                                          "nwCommand":"ANYTIME_FEEDBACK_RECIPIENT_SELECTION",
                                          "wid":"2a37f8012406102d1ad61154601e02a1",
                                          "descriptor":"Logan Johnson"
                                       }
                                    }
                                 ]
                              }
                           ]
                        }
                     ]
                  }
               ]
            }`

Which defines the column width for one of the columns as "automatic". While this is not a valid value for the width parameter, the card renders fine on iOS and Web, however it doesn't render on android.

As rendered on iOS
image

As rendered on Web
image

The web visualizer shows the message - Invalid column width:automatic - defaulting to "auto"

However android doesn't render the card at all.

### PR DESCRIPTION
**Pull request by almedina-ms**
## Related Issue
Fixes #3864 

## Description
Changes the behaviour for Column rendering when the width is not an expected value (auto, stretch, pixels in number or proportion). The new behaviour is defaulting to the same behaviour as "auto" width but a warning is lifted following the same behaviour of the UWP renderer

## How Verified
Unit tests were not able to be added as it is a rendering issue, not a parsing issue

The behaviour was tested manually and as can be seen below, the card renders properly

Screenshot_20200323-203239_AdaptiveCards Visualizer

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3868)

### CODE REVIEW

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ColumnRenderer.java [hunk 1]
```diff
@@ -156,30 +157,32 @@ else if ((column = Column.dynamic_cast(baseCardElement)) == null)
...
-            if (TextUtils.isEmpty(columnSize) || columnSize.equals(g_columnSizeStretch))
-            {
-                layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, LinearLayout.LayoutParams.MATCH_PARENT, 1);
-                returnedView.setLayoutParams(layoutParams);
-            }
-            else if (columnSize.equals(g_columnSizeAuto))
+            try
             {
-                layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.MATCH_PARENT);
+                // I'm not sure what's going on here
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ColumnRenderer.java:162]: > // I'm not sure what's going on here [](start = 15, length = 37)

Hmm this comment makes me feel uneasy.. What is going on with the comment here?!

**Comment by shalinijoshi19** [comment]
@almedina-ms we should also updated/add our documentation around the behavior around this API now. This should at least find its way into the release notes for starters and then have our Android documentation added.

File: source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ColumnRenderer.java [hunk 2]
```diff
@@ -156,30 +157,32 @@ else if ((column = Column.dynamic_cast(baseCardElement)) == null)
...
+                else
                 {
-                    throw new IllegalArgumentException("Column Width (" + column.GetWidth() + ") is not a valid weight ('auto', 'stretch', <integer>).");
+                    // If the width is Auto or is not valid (not weight, pixel, empty or stretch)
+                    layoutParams = new LinearLayout.LayoutParams(LinearLayout.LayoutParams.WRAP_CONTENT, LinearLayout.LayoutParams.MATCH_PARENT);
+                    returnedView.setLayoutParams(layoutParams);
+
+                    if (!columnSize.equals(g_columnSizeAuto))
+                    {
+                        renderedCard.addWarning(new AdaptiveWarning(AdaptiveWarning.INVALID_COLUMN_WIDTH_VALUE, "Column Width (" + column.GetWidth() + ") is not a valid weight ('auto', 'stretch', <integer>)."));
```
Comment (shalinijoshi19) [inline source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/readonly/ColumnRenderer.java:184]: Probably update the warning text to include what the behavior is defaulting to also for debuggability.

**Comment by shalinijoshi19** [comment]
@almedina-ms take a look at this PR please. I've also updated the release vehicle from 1.2.8 to 1.2.9 now. Thanks

**Comment by shalinijoshi19** [comment]
@almedina-ms build errors in UWP looks like? @paulcam206 FYI

**Comment by paulcam206** [comment]
see #4040 :)

**Comment by shalinijoshi19** [comment]
@almedina-ms is this ready to go to master and release/1.2 for the upcoming patch release?

**Pull request merged**